homeassistant/auth/mfa_modules/notify.py

```diff
@@ -98,15 +100,16 @@ def input_schema(self) -> vol.Schema:
 
     async def _async_load(self) -> None:
         """Load stored data."""
-        data = await self._user_store.async_load()
+        async with self._init_lock:
```

Comment:
Once inside the lock, the first thing to do is return if `self._user_settings is not None`.  Another call may populated it while it was waiting on the lock.